### PR TITLE
[Debt] Replace `pools` query with `poolsPaginated`

### LINKED ISSUE #9718: ♻️ Replace the `pools` query with a `poolsPaginated` query
## ♻️ Debt/Refactor

We have a lot of pools at this point, and the number will continue to grow. We should deprecate the `pools` query and replace it with a paginated version. 

The client should switch to using the paginated version under the hood, but this change should be invisible to users.

## 🕵️ Details

Note: `publishedPools` query will remain non-paginated for now.

## 🙋‍♀️ Proposed Solution

(optional) Do you have a proposed solution?

## 🌎 Localization

(optional) Provide any new copy along with translations available.

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the debt was addressed and expected functionality has not been affected.

- [ ] `pools` query deprecated and not used anywhere client-side
- [ ] use of `pools` query replaced with a `poolsPaginated` query

## 🛑 Blockers

Issues which must be completed before this one.

```[tasklist]
### Blocked By
```

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #9718

## 👋 Introduction

Replaces the existing `pools` query with a paginated version.

## 🧪 Testing

1. Build `pnpm run dev`
2. Login as admin `admin@test.com`
3. Navigate to areas affected:
    - Admin pools page
    - Pool candidate page (add to pool dialog, change status dialog)
    - User information page (add to pool dialog)
    - Pool candidates table (filter dialog)

### CODE REVIEW

File: api/app/Models/Pool.php [hunk 1]
```diff
@@ -314,8 +314,139 @@ public static function scopeCurrentlyActive(Builder $query)
         return $query;
     }
 
+    public static function scopeName(Builder $query, ?string $name): Builder
+    {
+        if ($name) {
+            $query->where(function ($query) use ($name) {
+                $term = sprintf('%%%s%%', $name);
```
Comment (tristan-orourke) [inline api/app/Models/Pool.php:321]: What exactly to the five % do? It thought we usually only have one on either side 😆

Comment (esizer) [reply api/app/Models/Pool.php:321]: with `sprintf` the `%` indicates a replacement value so you add two `%%` to escape it :sweat_smile:

Comment (esizer) [reply api/app/Models/Pool.php:321]: https://www.php.net/manual/en/function.sprintf.php#:~:text=With%20printf()%20and%20sprintf()%20functions%2C%20escape%20character%20is%20not%20backslash%20%27%5C%27%20but%20rather%20%27%25%27.

File: api/graphql/schema.graphql [hunk 2]
```diff
@@ -628,6 +628,21 @@ input UserPublicProfileFilterInput {
   publicProfileSearch: String @scope
 }
 
+input PoolFilterInput {
+  generalSearch: String @scope
+  name: String @scope
+  ownerName: String @scope
+  ownerEmail: String @scope
```
Comment (tristan-orourke) [inline api/graphql/schema.graphql:635]: We should probably deprecate these fields on the Pool, actually, and skip adding them here. I don't think we're using them anywhere.

**Review by tristan-orourke — COMMENTED** [review]
Could you add some phpunit tests testing the filters on the new pool query?

File: apps/web/src/pages/Pools/IndexPoolPage/components/helpers.tsx [hunk 3]
```diff
@@ -135,3 +147,118 @@ export function ownerNameAccessor(pool: Pool) {
...
+  return {
+    generalSearch: !!search.term && !search.type ? search.term : undefined,
+    name: search.type === "name" ? search.term : undefined,
+    ownerName: search.type === "ownerName" ? search.term : undefined,
+    ownerEmail: search.type === "ownerEmail" ? search.term : undefined,
+    team: search.type === "type" ? search.term : undefined,
+  };
+}
+
+export function getOrderByClause(
```
Comment (tristan-orourke) [inline apps/web/src/pages/Pools/IndexPoolPage/components/helpers.tsx:172]: Setting up this sorting is so much more complicated than I remembered 🫣

File: api/graphql/schema.graphql (hunk 2 above)
Comment (esizer) [reply api/graphql/schema.graphql]: Oh yeah, I was wondering if we needed this or not. Removed in [remove owner scopes from pool](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10383/commits/ba6cf3f8d9b866fb1bf7839fb32ab203ac4e8a04)

Just working on the tests now :+1:

File: apps/web/src/pages/Pools/IndexPoolPage/components/helpers.tsx (hunk 3 above)
Comment (esizer) [reply apps/web/src/pages/Pools/IndexPoolPage/components/helpers.tsx:170]: Oh yeah, thankfully we had an example I could copy and paste from :stuck_out_tongue:

File: api/graphql/schema.graphql (hunk 2 above)
Comment (tristan-orourke) [reply api/graphql/schema.graphql]: ownerName and ownerEmail I mean

File: apps/web/src/pages/Users/UserInformationPage/components/ChangeStatusDialog.tsx [hunk 4]
```diff
@@ -269,7 +268,13 @@ const ChangeStatusDialog = ({
...
               </p>
               <div data-h2-margin="base(x.5, 0, x.125, 0)">
-                <Combobox
+                <PoolFilterInput
+                  includeIds={userPoolIds}
+                  filterInput={{
+                    statuses: [PoolStatus.Closed, PoolStatus.Published],
+                  }}
+                />
+                {/* <Combobox
```
Comment (tristan-orourke) [inline apps/web/src/pages/Users/UserInformationPage/components/ChangeStatusDialog.tsx:277]: Is this meant to be removed?

File: api/graphql/schema.graphql (hunk 2 above)
Comment (esizer) [reply api/graphql/schema.graphql]: Oh, would you like them to be removed from the `PoolTable` as removing the scopes?

File: apps/web/src/pages/Users/UserInformationPage/components/ChangeStatusDialog.tsx (hunk 4 above)
Comment (esizer) [reply apps/web/src/pages/Users/UserInformationPage/components/ChangeStatusDialog.tsx]: Oops, yeah I did :sweat_smile: Removed in [remove commented combobox](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10383/commits/0124c4b3d14607d1c7bd0bc146347bc5f4603774)

**Comment by tristan-orourke** [comment]
@esizer could you take a look at the visual diffs on chromatic? Looks like some information is getting lost.

**Comment by esizer** [comment]
> @esizer could you take a look at the visual diffs on chromatic? Looks like some information is getting lost.

Okay, I think I was able to fix the request form (I forgot ot add the mock query). However, the table diff was because I couldn't get the sort on the status because it is not actually a database column. I may need assistance on that because with the paginated queries, we can only sort by selected database columns since it is just doing a SQL `ORDER BY`.

I tried adding the `addSelect` but that was overriding the other selects :confounded: 

[fix request form story](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10383/commits/3842c7deab36192c932f01a507679ca9e8d74401)

File: api/graphql/schema.graphql [hunk 5]
```diff
@@ -747,6 +760,26 @@ type Query {
...
+        relations: [
+          { relation: "user", columns: ["first_name", "email"] }
+          { relation: "classification", columns: ["group", "level"] }
+        ]
+      )
+  ): [Pool]!
+    @paginate(
+      defaultCount: 10
+      maxCount: 1000
+      scopes: ["authorizedToView", "notArchived"]
```
Comment (esizer) [inline api/graphql/schema.graphql:780]: Working on the tests and was like "why is the archived scope failing?!" Then I remembered this...

Do we want this? :thinking: Or should I just not consider archived in the tests? @tristan-orourke

**Comment by esizer** [comment]
> Could you add some phpunit tests testing the filters on the new pool query?

Added tests and the bugs it uncovered :stuck_out_tongue: [add tests, fix bugs from tests](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10383/commits/17fe2b71038821023abe405a503f0ba2aa8284dc)

File: api/graphql/schema.graphql (hunk 5 above)
Comment (tristan-orourke) [reply api/graphql/schema.graphql:780]: We can take off the built-in notArchived scope, though most places querying pools will want to add it. Could we make it enabled by default?

Comment (esizer) [reply api/graphql/schema.graphql:780]: Hmm, I'll take a look. I'm pretty sure you can provide defaults and then when you pass something in, it overrides it :thinking:

Comment (tristan-orourke) [reply api/graphql/schema.graphql:780]: I don't think there's an easy way to make it default. That's fine.

**Review by tristan-orourke — CHANGES_REQUESTED** [review]
Found a couple funny things on the pools table.

If I filter by Publishing Group or Team, then the page controls don't notice the reduced results. (Is this filtering happening on the frontend?) 
image

Meanwhile, you can't search by publishing group unless you select the "Filter by Publishing Group" name, because the general search scope doesn't include it.

**Comment by esizer** [comment]
> If I filter by Publishing Group or Team, then the page controls don't notice the reduced results. (Is this filtering happening on the frontend?)

Oh yeah, I actually did not intended to allow searching by publishing group since it is tricky due to being an enum and will be moving to filters once this gets merged. Does that sound right, or do you want me to try and figure out searching by enum value through the DB? :thinking:

**Comment by tristan-orourke** [comment]
You're right, publishing group should be handled by filters, not the search bar. (Which is why we have https://github.com/GCTC-NTGC/gc-digital-talent/issues/6169!) Is there anything we can do to mitigate the buggyness in the meantime? 🤔

**Comment by esizer** [comment]
> You're right, publishing group should be handled by filters, not the search bar. (Which is why we have #6169!) Is there anything we can do to mitigate the buggyness in the meantime? 🤔

I can just remove it from the search bar in this PR. Then, I'll pick the filter one up right away so we don't lose that ability in between deploys. Does that sound good?

**Review by tristan-orourke — APPROVED** [review]
That sounds great @esizer

**Pull request merged**